perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/ConsentsManagerBlImpl.java

```diff
@@ -38,6 +51,112 @@ public void setPerunBl(PerunBl perunBl) {
 		this.perunBl = perunBl;
 	}
 
+
+	public Consent createConsent(PerunSession sess, Consent consent) throws ConsentExistsException, UserNotExistsException, PrivilegeException, ConsentHubNotExistsException, ConsentNotExistsException {
+		Utils.notNull(consent, "consent");
+
+		ConsentHub consentHub = getConsentHubById(sess, consent.getConsentHub().getId());
+		if (consentHub == null) {
+			throw new ConsentHubNotExistsException("Consent hub with id=" + consent.getConsentHub().getId() + " doesn't exist.");
+		}
+
+		// Add attributes from ConsentHub
+		List<AttributeDefinition> filter = consent.getAttributes();
+		for (Facility facility : consentHub.getFacilities()) {
+			for (Service service : perunBl.getServicesManagerBl().getAssignedServices(sess, facility)) {
+				for (AttributeDefinition attr : perunBl.getAttributesManagerBl().getRequiredAttributesDefinition(sess, service)) {
+					if (checkAttributeConsent(attr)) {
+						filter.add(attr);
+					}
+				}
+			}
+		}
+		consent.setAttributes(filter);
+
+		consent = getConsentsManagerImpl().createConsent(sess, consent);
+		getPerunBl().getAuditer().log(sess, new ConsentCreated(consent));
+
+		return consent;
+	}
+
+	/*
+	 * Checks if attribute should be added
+	 */
+	private boolean checkAttributeConsent(AttributeDefinition attribute) {
+		String attributeNamespace = attribute.getNamespace();
+		return attributeNamespace.startsWith(AttributesManager.NS_MEMBER_RESOURCE_ATTR) ||
+			attributeNamespace.startsWith(AttributesManager.NS_MEMBER_GROUP_ATTR) ||
+			attributeNamespace.startsWith(AttributesManager.NS_MEMBER_ATTR_CORE) ||
+			attributeNamespace.startsWith(AttributesManager.NS_USER_FACILITY_ATTR) ||
+			attributeNamespace.startsWith(AttributesManager.NS_USER_ATTR) ||
+			attributeNamespace.startsWith(AttributesManager.NS_UES_ATTR);
+	}
+
+
+	public void deleteConsent(PerunSession sess, Consent consent) throws ConsentNotExistsException {
+		Utils.notNull(consent, "consent");
+		Utils.notNull(consent.getId(), "consent.id");
```

Comment:
We can omit this check because id is a primitive integer and thus never a null :)